pkg/frontend/middleware/log.go

```diff
@@ -52,9 +52,18 @@ func Log(baseLog *logrus.Entry) func(http.Handler) http.Handler {
 			r.Body = &logReadCloser{ReadCloser: r.Body}
 			w = &logResponseWriter{ResponseWriter: w, statusCode: http.StatusOK}
 
+			// correlationID is set by arm frontend. Local RP will not have this set
 			correlationID := r.Header.Get("X-Ms-Correlation-Request-Id")
```

Comment:
@jim-minter I assume `X-Ms-Correlation-Request-Id` one is set by ARM frontend, as in our current local dev its `""`. Should we set this if not set?

`X-Ms-Request-Id` is set by SDK, so we should generate only if API consumer is not standard and this field is empty?
WDYT?

Reply:
x-ms-correlation-request-id: set by arm, don't set it, forward to backend for logging
x-ms-request-id: set by us, forward to backend for logging
x-ms-client-request-id: optionally set by client, forward to backend for logging

see https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-details.md

Reply:
ok, so 3 fields to forward.  Maybe this?
```
doc.Metadata{
CorrelationRequestID
RequestID
ClientRequestID
}
```